docs/docs/task-handling.md

```diff
@@ -56,6 +56,13 @@ case TASK_KILLED => Killed
 ```
 The task was killed by the executor.
 
+
+**Note:** Mesos also sends a deprecated `TASK_LOST` message in certain situations. A future version of Mesos will no longer send this message. You may receive a `TASK_LOST` message if a Mesos agent is removed because a new Mesos agent registers with the same IP address. When a `TASK_LOST` message is received, Marathon will try to get more information about the reason the message was sent.
```

Comment:
> You may receive a `TASK_LOST` message if a Mesos agent is removed because a new Mesos agent registers with the same IP address.

_Marathon_ may receive

We should double check with the Mesos team if the same IP address is pertinent; it's my assumption. I'm not sure what would happen if the host reboots and comes back before the timeout situation, and has a different IP address.

> When a `TASK_LOST` message is received, Marathon will try to get more information about the reason the message was sent.

Marathon will look at the TASK_LOST reason (this is sent along with the status update) and use it to infer `TASK_GONE`, `TASK_UNREACHABLE`, or `TASK_UNKOWN`.

I am not currently certain which reason they send in the case of a slave removal.